x/concentrated-liquidity/fees_test.go

```diff
@@ -1015,3 +1020,152 @@ func (s *KeeperTestSuite) TestPreparePositionAccumulator() {
+		s.Require().NoError(err)
+		totalFeesCollected = totalFeesCollected.Add(coins...)
+		s.tickStatusInvariance(activeTicks, DefaultMinTick, DefaultMaxTick, coins, expectedFeeDenoms[0])
+	}
+
+	// Claim narrow range position fees across three of four accounts
+	for i := 0; i < 3; i++ {
+		coins, err := s.App.ConcentratedLiquidityKeeper.CollectFees(ctx, poolId, s.TestAccs[i], DefaultLowerTick, DefaultUpperTick)
+		s.Require().NoError(err)
+		totalFeesCollected = totalFeesCollected.Add(coins...)
+		s.tickStatusInvariance(activeTicks, DefaultLowerTick, DefaultUpperTick, coins, expectedFeeDenoms[1])
+	}
+
+	// Claim consecutive range position fees across two of four accounts
+	for i := 0; i < 2; i++ {
+		coins, err := s.App.ConcentratedLiquidityKeeper.CollectFees(ctx, poolId, s.TestAccs[i], DefaultExponentConsecutivePositionLowerTick.Int64(), DefaultExponentConsecutivePositionUpperTick.Int64())
+		s.Require().NoError(err)
+		totalFeesCollected = totalFeesCollected.Add(coins...)
+		s.tickStatusInvariance(activeTicks, DefaultExponentConsecutivePositionLowerTick.Int64(), DefaultExponentConsecutivePositionUpperTick.Int64(), coins, expectedFeeDenoms[2])
+	}
+
+	// Claim overlapping range position fees on one of four accounts
+	coins, err := s.App.ConcentratedLiquidityKeeper.CollectFees(ctx, poolId, s.TestAccs[0], DefaultExponentOverlappingPositionLowerTick.Int64(), DefaultExponentOverlappingPositionUpperTick.Int64())
+	s.Require().NoError(err)
+	totalFeesCollected = totalFeesCollected.Add(coins...)
+	s.tickStatusInvariance(activeTicks, DefaultExponentOverlappingPositionLowerTick.Int64(), DefaultExponentOverlappingPositionUpperTick.Int64(), coins, expectedFeeDenoms[3])
+
+	// Define error tolerance
+	var errTolerance osmomath.ErrTolerance
+	errTolerance.AdditiveTolerance = sdk.NewDec(10)
+	errTolerance.RoundingDir = osmomath.RoundDown
+
+	// Check that the total fees collected is equal to the total fees (within a tolerance)
+	for _, coin := range totalFeesCollected {
+		expected := totalFees.AmountOf(coin.Denom)
+		actual := coin.Amount
+		s.Require().Equal(0, errTolerance.Compare(expected, actual), fmt.Sprintf("expected (%s), actual (%s)", expected, actual))
+	}
+}
+
+// tickStatusInvariance tests if the swap position was active during the given tick range and
+// checks that the fees collected are non-zero if the position was active, or zero otherwise.
+func (s *KeeperTestSuite) tickStatusInvariance(ticksActivatedAfterEachSwap [][]sdk.Int, lowerTick, upperTick int64, coins sdk.Coins, expectedFeeDenoms []string) {
+	var positionWasActive bool
+	// Check if the position was active during the swap
+	for i, ticks := range ticksActivatedAfterEachSwap {
+		for _, tick := range ticks {
+			if tick.GTE(sdk.NewInt(lowerTick)) && tick.LTE(sdk.NewInt(upperTick)) {
+				positionWasActive = true
+				break
+			}
+		}
+		if positionWasActive {
+			// If the position was active, check that the fees collected are non-zero
+			if expectedFeeDenoms[i] != "" {
```

Comment:
Suggest defining a constant for an empty string. It will make it easier to identify where and why we set it to empty string

Reply:
https://github.com/osmosis-labs/osmosis/pull/4282/commits/c1a61b883c5abdc399a11108c5fc05d65a6083a8